### ISSUE TITLE
[BUG] v2 data engine: volume.snapshotDelete leaves snapshot in incorrect state and volume cannot detach

### ISSUE THREAD

**Issue by chriscchien** [longhorn/longhorn#12395]
### Describe the Bug

When calling `volume.snapshotDelete` on a v2 data engine volume, the snapshot deletion result is incorrect and causes the volume to be unable to detach.

**v1 Data Engine Result (Expected)**
```
# kl get snapshot
NAME                    VOLUME                    CREATIONTIME           READYTOUSE   RESTORESIZE   SIZE   AGE
snap-4044d6d5fa06495d   longhorn-testvol-grydmc   2025-12-29T08:29:59Z   false        33554432      0      6m27s
```
**v2 Data Engine Result (Incorrect)**
According to the documentation, when using the v2 data engine, the Snapshot CR should be removed immediately after deletion:
https://longhorn.io/docs/1.11.0/snapshots-and-backups/setup-a-snapshot/#data-engine-behavioral-differences

However, the Snapshot CR still exists, and its fields are incomplete:
```
# kl get snapshot
NAME                    VOLUME                    CREATIONTIME   READYTOUSE   RESTORESIZE   SIZE   AGE
snap-4a850d668ce74343   longhorn-testvol-osyl2x                                                    2m15s
```
After this state occurs, the volume cannot be detached and longhorn manager have logs for parse last healthy timestamp fail.
```
time="2025-12-29T08:43:30.370594Z" level=warning msg="Failed to check if replica longhorn-testvol-osyl2x-r-7824ca06 transitioned to mode RW after it was last healthy" func="controller.(*VolumeController).ReconcileEngineReplicaState" file="volume_controller.go:754" accessMode=rwo controller=longhorn-volume currentEngine=longhorn-testvol-osyl2x-e-0 error="cannot parse timestamp : parsing time \"\" as \"2006-01-02T15:04:05Z07:00\": cannot parse \"\" as \"2006\"" frontend=blockdev migratable=false node=ip-10-0-2-226 owner=ip-10-0-2-226 state=attached volume=longhorn-testvol-osyl2x
```

### To Reproduce

1. In longhorn-test pod run `export RUN_V2_TEST=true`
2. Run below test case
```python
def test_snapshot_test1(client, volume_name, backing_image=""):  # NOQA
    volume = create_and_check_volume(client, volume_name,
                                     backing_image=backing_image)

    lht_hostId = get_self_host_id()
    volume = volume.attach(hostId=lht_hostId)
    volume = common.wait_for_volume_healthy(client, volume_name)

    volume = client.by_id_volume(volume_name)

    snap1 = create_snapshot(client, volume_name)
    
    volume.snapshotDelete(name=snap1.name)
    
    volume.detach()
    volume = common.wait_for_volume_detached(client, volume_name)
```
3. test case stuck at `volume.detach()` and meanwhile, `kubectl -n longhorn-system get snapshot` can see a snapshot CR remain without `CREATIONTIME`   `READYTOUSE`   `RESTORESIZE`

### Expected Behavior

The snapshot CR delete by `volume.snapshotDelete` and volume can detach

### Support Bundle for Troubleshooting

[supportbundle_92bc4844-42ce-4ed9-b886-83a8eaeb8745_2025-12-29T08-48-52Z.zip](https://github.com/user-attachments/files/24365687/supportbundle_92bc4844-42ce-4ed9-b886-83a8eaeb8745_2025-12-29T08-48-52Z.zip)

### Environment

- Longhorn version: master-head
- Impacted volume (PV): longhorn-testvol-osyl2x
- Installation method (e.g. Rancher Catalog App/Helm/Kubectl): kubectl
- Kubernetes distro (e.g. RKE/K3s/EKS/OpenShift) and version:
  - Number of worker nodes in the cluster: 6.14.0-1018-aws
- Node config
  - OS type and version: ubuntu 24.04
  - Kernel version:

**Comment by chriscchien** [comment]
Using `volume.snapshotCRDelete` instead of using `volume.snapshotDelete` introduced in  https://github.com/longhorn/longhorn-tests/pull/2980 resolved this issue

**Comment by innobead** [comment]
> Using `volume.snapshotCRDelete` instead of using `volume.snapshotDelete` introduced in [longhorn/longhorn-tests#2980](https://github.com/longhorn/longhorn-tests/pull/2980) resolved this issue

I think some methods here should be deprecated or even removed, since we won't allow users to operate snapshots directly but rather rely on snapshot CR. cc @derekbit @houhoucoop 

/home/dko/Pictures/Screenshots/Screenshot From 2025-12-29 17-20-28.png

**Comment by derekbit** [comment]
Duplicate https://github.com/longhorn/longhorn/issues/12394
